x/emissions/proto/emissions/v9/events.proto

```diff
@@ -436,3 +434,146 @@ message EventPruneRecords {
   int64 block_height = 1;
   uint64 topic_id = 2;
 }
+
+// Stake removal processing events
+message EventReputerStakeRemovalCompleted {
+  uint64 topic_id = 1;
+  string reputer = 2;
+  string amount = 3 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  int64 block_height = 4;
+}
+
+message EventDelegateStakeRemovalCompleted {
+  uint64 topic_id = 1;
+  string reputer = 2;
+  string delegator = 3;
+  string amount = 4 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  int64 block_height = 5;
+}
+
+message EventDelegateRewardShareUpdated {
+  uint64 topic_id = 1;
+  string reputer = 2;
+  string reward_per_share = 3 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  int64 block_height = 4;
+}
+
+message EventDelegateRewardDistributed {
+  uint64 topic_id = 1;
+  string reputer = 2;
+  string amount = 3 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  int64 block_height = 4;
+}
+
+message EventActiveReputersSet {
+  uint64 topic_id = 1;
+  repeated string addresses = 2;
+  int64 nonce_block_height = 3;
+}
+
+message EventActiveInferersSet {
+  uint64 topic_id = 1;
+  repeated string addresses = 2;
+  int64 nonce_block_height = 3;
+}
+
+message EventActiveForecastersSet {
+  uint64 topic_id = 1;
+  repeated string addresses = 2;
+  int64 nonce_block_height = 3;
+}
+
+message EventActiveTopicsAtBlockSet {
+  int64 target_block_height = 1;
+  repeated uint64 topic_ids = 2;
+}
```

Comment:
What's the value of exposing that as an event? Capturing nonce related events seems enough to track epochs lifecycle